### PR TITLE
Omit not used runtime in flat packer

### PR DESCRIPTION
**Pull request by TrySound**

### CODE REVIEW

File: Fastpack/FlatPacker.ml [hunk 1]
```diff
@@ -854,22 +858,28 @@ let pack ?(cache=Cache.fake ()) (ctx : Context.t) channel =
...
-  let mode =
-    Printf.sprintf "var __DEV__ = %s;\n"
-      @@ if ctx.mode = Mode.Development then "true" else "false"
+  let%lwt () = pack ctx MDM.empty in
+
+  let bundle = channel
+    |> Lwt_io.position
+    |> Int64.to_int
+    |> Lwt_bytes.extract bytes 0
+    |> Lwt_bytes.to_string
```
Comment (andreypopp) [inline Fastpack/FlatPacker.ml]: What's the perf implication of this?

Comment (TrySound) [reply Fastpack/FlatPacker.ml]: @andreypopp Functions are hoisted, variables too, but `undefined`.

Comment (zindel) [reply Fastpack/FlatPacker.ml]: @TrySound if it is only `import()` calls - it should be fine. It only executes in `Promise.resolve` hence function `__fastpack_import__` will be evaluated at that time

File: test/flat-packer-base/__snapshots__/flat-packer.test.js.snap [hunk 2]
```diff
@@ -195,6 +141,123 @@ function __fastpack_import__(f) {
...
+return $n__dep$$_$$esm$$_$$named.exports;
+}
+"
+`;
+
+exports[`flat packer with dynamic esm dep: target: esm 1`] = `
+"var __DEV__ = false;
+var __fastpack_cache__ = {};
+
+function __fastpack_require__(f) {
```
Comment (zindel) [inline test/flat-packer-base/__snapshots__/flat-packer.test.js.snap]: `__fastpack_require__` seems to be redundant. We have treat all `require()` calls as static dependencies at the moment. Maybe we could rip it off entirely?

File: Fastpack/FlatPacker.ml (hunk 1 above)
Comment (TrySound) [reply Fastpack/FlatPacker.ml]:
```
$ ./fpack_master src/index.js
Packed in 3.041s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_master src/index.js
Packed in 3.069s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_master src/index.js
Packed in 3.048s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_master src/index.js
Packed in 3.086s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_new src/index.js
Packed in 3.051s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_new src/index.js
Packed in 3.100s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_new src/index.js
Packed in 3.058s. Number of modules: 808. Cache: no. Mode: production

$ ./fpack_new src/index.js
Packed in 3.229s. Number of modules: 808. Cache: no. Mode: production
```

**Pull request merged**